lib/src/revset.rs

```diff
@@ -2065,18 +2065,47 @@ impl VisibilityResolutionContext<'_> {
     }
 }
 
+pub trait RevsetIterator<T>: Iterator<Item = Result<T, RevsetEvaluationError>> {
+    fn unwrapped<'a>(self: Box<Self>) -> Box<dyn Iterator<Item = T> + 'a>
+    where
+        Self: 'a,
+        T: 'a;
+}
+
+impl<T, I: Iterator<Item = Result<T, RevsetEvaluationError>>> RevsetIterator<T> for I {
+    fn unwrapped<'a>(self: Box<Self>) -> Box<dyn Iterator<Item = T> + 'a>
+    where
+        Self: 'a,
+        T: 'a,
+    {
+        Box::new(UnwrappedRevsetIterator { iter: self })
```

Comment:
It's bad to suggest using `.unwrap()`. Maybe we can use `.try_collect().unwrap()`, `.map_ok()`, etc. instead?
https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.map_ok

Reply:
I inlined `map(Result::unwrap)` at the call sites instead.